cmd/argocd-k8s-auth/commands/token_file.go

```diff
@@ -0,0 +1,56 @@
+package commands
+
+import (
+	"fmt"
+	"os"
+	"time"
+
+	"github.com/argoproj/argo-cd/v2/util/errors"
+	"github.com/golang-jwt/jwt/v4"
+	"github.com/spf13/cobra"
+)
+
+// newTokenFileCommand returns a new instance of a command that uses local token from the file
+// See https://kubernetes.io/docs/concepts/storage/projected-volumes/#serviceaccounttoken
+func newTokenFileCommand() *cobra.Command {
+	var (
+		file string
+	)
+	var command = &cobra.Command{
+		Use: "token-file",
+		Run: func(c *cobra.Command, args []string) {
+			token, err := getTokenFile(file)
+			errors.CheckError(err)
+			_, _ = fmt.Fprint(os.Stdout, token)
+		},
+	}
+	command.Flags().StringVar(&file, "file", "/var/run/secrets/kubernetes.io/serviceaccount/token", "Path to a token file")
+	return command
+}
+
+func getTokenFile(path string) (string, error) {
+	tokenBuf, err := os.ReadFile(path)
+	if err != nil {
+		return "", err
+	}
+
+	tokenStr := string(tokenBuf)
+
+	token, _, err := new(jwt.Parser).ParseUnverified(tokenStr, jwt.MapClaims{})
+	if err != nil {
+		return "", err
+	}
+
+	claims, ok := token.Claims.(jwt.MapClaims)
+	if !ok {
+		return "", fmt.Errorf("Error extracting claims from token %s", path)
+	}
+
+	exp, ok := claims["exp"].(float64)
```

Comment:
Would it make sense to be able to check the audience (`aud`) claim if a command line argument is given?

Reply:
I am not sure I understand the reasoning, but I am not an expert in OIDC specification, so I might be missing something. From what I do understand - it is a job of the relying party to look into `aud` and prevent confused deputy attack if the token was not addressed to it by the IdP. ArgoCD in this scenario is just a "user" and it has no business doing anything with the token other than holding it. Can you point to the spec or provide an example of the attack that this check would be preventing?

Reply:
TBH, I was not thinking of any specific attacks; it was more along the lines of having an extra (optional) assertion to prevent the accidental usage of a token which is not intended for Argo CD. I'm probably thinking about this the wrong way, so nevermind 😄

Reply:
As a matter of fact, specification seem to permit for the token to be encrypted end to end (idp to relying party), so that the bearer (user, argocd) can't even look into it. I am not aware of any particular k8s specific issuer implementations that does that today, but it is in the spec. So I'd vote to not to over engineer it and keep future-compatible.